pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -854,45 +838,6 @@ func (p *Provisioner) validateDriver(cluster *apimgmtv3.Cluster) (string, error)
 	return newDriver, nil
 }
 
-func (p *Provisioner) getSystemImages(spec apimgmtv3.ClusterSpec) (*rketypes.RKESystemImages, error) {
-	version := spec.RancherKubernetesEngineConfig.Version
-	if version == "" {
-		return nil, fmt.Errorf("kubernetes version (spec.rancherKubernetesEngineConfig.kubernetesVersion) is unset")
-	}
-	// fetch system images from settings
-	systemImages, err := kd.GetRKESystemImages(version, p.RKESystemImagesLister, p.RKESystemImages)
-	if err != nil {
-		return nil, fmt.Errorf("failed to find system images for version %s: %v", version, err)
-	}
-
-	privateRegistry := util.GetPrivateRegistryURL(&apimgmtv3.Cluster{Spec: spec})
-	if privateRegistry == "" {
-		return &systemImages, nil
-	}
-
-	// prepend private repo
-	imagesMap, err := convert.EncodeToMap(systemImages)
-	if err != nil {
-		return nil, err
-	}
-	updatedMap := make(map[string]interface{})
-	for key, value := range imagesMap {
-		newValue := fmt.Sprintf("%s/%s", privateRegistry, value)
-		updatedMap[key] = newValue
-	}
-	// Decoding updateMap to systemImages using json marshal/unmarshal to honor field names
-	updatedByte, err := json.Marshal(updatedMap)
-	if err != nil {
-		return nil, err
-	}
-	err = json.Unmarshal(updatedByte, &systemImages)
-	if err != nil {
-		return nil, err
-	}
-	logrus.Debugf("Updated system images to use private registry [%s]: %#v", privateRegistry, systemImages)
-	return &systemImages, nil
-}
-
 // getSpec computes the spec from the given cluster. Returns an error if the spec could not be computed. Otherwise,
 // returns the spec and a boolean that will be true if the cluster config has changed and false otherwise.
 func (p *Provisioner) getSpec(cluster *apimgmtv3.Cluster) (spec *apimgmtv3.ClusterSpec, configChanged bool, err error) {
```

Comment:
can `reconcileRKENodes` be removed as well?

Reply:
Removed here: https://github.com/rancher/rancher/pull/50599/files#diff-b492a84101164ca8ea03ff0154d84dd1bf90b862bcf6fd6cf63de035d0dae757L904